### ISSUE TITLE
bug: 2FSA version 5.4.8 does not save backup file

### ISSUE THREAD

**Issue by TenZeroOne** [twofas/2fas-android#177]
### Bug type

Other

### App version

2FSA version 5.4.8

### Device environment

Android Version 12
Model Galaxy A12 

### Bug description

After updating 2FSA to version 5.4.8 I am unable to "Export to a file" anymore. Upon saving my keys, I have a generic "Unknown error. Try again". File created has 0 bit. I tried several folders, even create a new one. I did try on SDD Card too with no avail. I did also check 2FSA permissions via Settings.

Note: I did reboot the device. I did try to write to any folder using another app: no issues. I can save pictures, videos etc: no issues 

**The other phone Oppo with Android 10 which runs 2FSA version 5.4.6 has not this issue** . . . and I will not update it because I fear I will not be able to backup my data.

Should you ask: I do not have Google drive, so I did not test a backup with it. I do not want to use one because I do not trust it (sorry to be blunt . . . LOL)  

### Acknowledgements

- [X] This issue is not a duplicate of an existing bug report.
- [X] I understand that security vulnerabilities should be reported to security@2fas.com instead of on GitHub.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by s3rious0** [comment]
Hey @TenZeroOne,


thank you to bring the Issue to us. 

I will try to help you with your local 2FAS Backup.

I have more questions to you:

- Have work the backups before you have make the update to version 5.4.8?

- Ist the error on both local backup variations persist? Backup with password and without password?

- Which Device, Android Version and Store ( e.g. Play store ), you are used?


On my Device ( Google Pixel / Android 15 ), I can't reproduce the error. On me, it works all fine.

Thank you.

Greeting.

**Comment by TenZeroOne** [comment]
Hi @s3rious0 

Thank you for your reply.

On your questions:

- Yes the backup was working before the update. I still have an old January 2024 backup file in the My_Backup folder on the SDD Card (I do backup tokens regularly)

- I did try that, sorry I should have mentioned it in my original post, to dump the backup with a password and without a password, both end up with the same "Unknown error. Try again" message.

- **The device I use is a Galaxy A12 - Model Number SM-A127F/DS - Running Android 12**. The 2FSA was downloaded and late updated from Google Store.

To verify whether it is my handset or device model or version of the software, I download 2FAS into two alternative devices:

- A very old hand set: Samsung SM-J500
- A Samsung Tablet SM-T500

**On both devices 2FSA works like magic.** I can import, export copy QR Code from Google Authenticator. I can even move the file via FTP from one device to another and import (overwrite) the tokens.

`So the issue lies either on my Device Model or the Device + Android Version or Device + Android Version + 2FA Version.`

I will uninstall then try to reinstall 2FSA because as it is, it remains useless for me . . . wish me luck

It would be nice to know whether other users experience the same issue.

**Comment by TenZeroOne** [comment]
**Update.**

- I was able to use "Export to File --> Share" via an App
- Uploaded the file into another device: those missing Tokens were added

**then**

- I wiped the 2FSA storage 2FSA --> Storage --> Clear Data
- I uninstalled 2FSA
- I opened Google Player
- I searched and installed 2FSA

**then**

- I imported from Google Authenticator one token
- I tried to export to a file ---> "Unknown error. Try again" message.

So it is not my handset but definitely a combination of:

`Handset Model + Android Version **or** Handset Model + Android Version + 2FA Version`

**Comment by s3rious0** [comment]
Hey @TenZeroOne, 


thank you for your detailed answers.

Have you made in the last time an software Update from Samsung?

When yes, please give me the Version Number of this update. It's likely most monthly updates and it names like e.g. " Android-Security-Update- 5. November 2024 ".

Then I take a look at this.

Eventually it is an bug in your android Version. 

Thank you.

Greetings.

**Comment by TenZeroOne** [comment]
Hello @s3rious0 

Update? And break another phone? I rather not have 2FSA then.

Last time I update its 8 month old brother Galaxy A12 I had to scrap it and buy a new handset. Basically the handset after the update was unusable, slow, not charging, WiFi disconnections . . . nope big nope, no updates.

Furthermore it dos not justify and address the **"why 2FSA works on his grand-father handset Samsung SM-J500, I bought  in August 2016"?**

**Comment by s3rious0** [comment]
Hey @TenZeroOne,


thank you for the answer.

I asked about the updates because new bugs can also occur in newer updates. However, if you have not carried out any updates in the last few months, it is probably not due to the Android version. 

The problem is more likely to come from something else.

Have you tried to install the previous version and test the described problems? ( Version 5.4.7 )

Thank you.

Greeting.

**Comment by TenZeroOne** [comment]
Good point! I will download the APK from the GitHub repo and I will try

**Comment by s3rious0** [comment]
Hi @TenZeroOne,

does it work with the version "5.4.7"?

Greeting
